common/src/main/java/gov/cms/ab2d/common/service/UserServiceImpl.java

```diff
@@ -34,12 +36,30 @@ public User getCurrentUser() {
 
     @Override
     public User getUserByUsername(String username) {
-        return userRepository.findByUsername(username);
+        User user = userRepository.findByUsername(username);
+        if (user == null) {
+            String userNotPresentMsg = "User is not present in our database";
+            log.error(userNotPresentMsg);
+            throw new ResourceNotFoundException(userNotPresentMsg);
+        } else {
+            return user;
+        }
     }
 
     @Override
-    public User createUser(UserDTO userDTO) {
```

Comment:
Should both of these public methods (create & update) have an `@Transactional` annotation to make sure all the queries happen within the same transaction?

Reply:
They happen during separate API calls.

Reply:
Actually I thought the service had one, I can put it there